.github/mergify.yml

```diff
@@ -24,19 +24,29 @@ queue_rules:
 
 # NOTE: in case of flaky tests you above checks will fail
 # the PR will be removed from the queue.
-# This retries the checks, as soon as they become green,
-# the PR will be pusehd to the queu again
+# Once the PR is ready to be merged the flaky tests will
+# be retried automatically.
+# As soon as the CI is green, the PR will be pusehd to the queu
 pull_request_rules:
-  - name: retry CI on fails (used for flaky tests)
+  - name: retry flaky thests (when PR is approved)
     conditions:
       - label=🤖-automerge
       - label!=🤖-do-not-merge
       - base=master
+
+      - -draft # PR is not in draft state
+      - -conflict # No merge conflicts
+      - "#approved-reviews-by>=2" # Requires 2 approving reviews
+      - "#changes-requested-reviews-by=0" # No changes requested
+      - "#review-threads-unresolved=0" # All review threads resolved
```

Comment:
why do you need to define all these things? can't it just check the github PR status?
I find it annoying as these settings are typically defined at the github repository level. now this needs to be duplicated in this file? Actually this also includes draft and conflict. In that case Github would anyway prevent you from merging. can this thing obliterate that fact??? if yes that looks dangerous. if not then I don't see the point in duplicating these settings.

Also other question:
- what about hotfix branches?

Reply:
This configuration file is not that clear. Unfortunately I also got very confused on how to use it at first.


This section `pull_request_rules` has nothing to do with the "Merge Queue" (section `queue_rules`). This part instructs the "Workflow Automation" part of Mergify to restart PRs whose tests have failed.
The "Merge Queue" part already respects and enforces all checks setup in GitHub.
The "Workflow Automation" does not know anything about these rules, so as a good measure when restarting the CI jobs (due to flaky tests) we check that the PR is ready to be merged.

Reply:
For now I want to try this only on the PRs but it can be added to all sorts of branches even hot fix ones. We can have a regex on the name of the branch and if it's a hotfix we let the "Workflow Automation" to restart it.

Reply:
ok. but still I don't get why you need to put all the criteria like "approved-reviews-by>=2" for example. this is setup in the settings of github. I don't get why this needs to be there.